src/actions/utils.cr

```diff
@@ -0,0 +1,79 @@
+module DppmRestApi::Actions
+  # A series of utility functions that are useful to all of the Action blocks
+  module Utils
+    macro fmt_route(route = "", namespace = false)
+    {{ "/" + @type.stringify
+         .downcase
+         .gsub(/^DppmRestApi::Actions::/, "")
+         .gsub(/::/, "/") }} {% if namespace %} + "/:namespace" {% end %} + ( {{route}} || "")
+    end
+
+    # the "last" or -1st block argument is selected because context is always the
+    # argument -- either the only or the second of two
+    {% for method in %w(get post put delete ws options) %}
+    macro relative_{{method.id}}(route, &block)
+      {{method.id}} fmt_route(\{{route}}) do |\{{block.args.splat}}|
+        namespace = \{{block.args[-1]}}.params.query["namespace"]? || DPPM::Prefix.default_group
+        \{{block.body}}
+      end
+    end
+    {% end %}
+
+    # Pull the context from the context's query parameter or use the default
+    # provided by the CLI.
+    #
+    # This macro only works inside of a route block with the variable "context"
+    # available
+    def get_prefix_or_default(from context)
+      DPPM::Prefix.new ensure_dir_exists context.params.query["prefix"]? || DPPM.default_prefix
```

Comment:
You can use `Prefix#create`, that ensures the directories exist. `ensure_dir_exists` can then be removed.
As you said, this is temporary.